### DISCUSSION TITLE
Keypad setup

### DISCUSSION THREAD

**Discussion by nirmalbaid-visa** [kuestess/homebridge-platform-insteonlocal#321] [General]
Hi 

Homebridge and insteonlocal development has been a lifesaver - I just got my set up configured. I am somewhat new to Insteon set up on Homebridge/Insteonlocal... 

I got all the lights and plugs working in my setup. I just added an 8-button keypad. I am not fully familiar with setting up the keypad in Homebridge. I linked the keypad device (3D7D33) using Insteonlocal UI and then added the config as below --

    "name": "Porch Lights",
    "deviceID": "3D7D33",
    "keypadbtn": "ON",
     "dimmable": "true",
     "deviceType": "keypad"

The Keypad is connected to the porch lights. When I press the On/Off button (A Button) the turn on / off but they dont work thru homebridge or homekit. Pls help.

thanks in advance.

**Comment by kuestess** [comment]
@nirmalbaid-visa You're mixing the 6 and 8 button nomenclature in your config.  The 'ON' value for the button is for 6-key keypads.  If you change `"keypadbtn": "ON"` to `"keypadbtn": "A"` it should work fine.  Alternately, assuming that this is the directly connected load (the default), you can just configure as a switch and that should work as well.

**Reply by nirmalbaid-visa** [reply]
Thanks - setting up as dimmer for the direct load worked !!
